debug_toolbar/panels/sqlalchemy.py

```diff
@@ -40,7 +41,13 @@ def after_execute(self, context: ExecutionContext, **kwargs: t.Any) -> None:
         }
         self.add_query(str(context.engine.url), query)
 
-    async def add_engines(self, request: Request):
+    async def add_engines(self, request: Request):  # noqa: C901
+        def add_bind_to_engines(bind: t.Union[Connection, Engine]):
```

Comment:
We can add a class method `add_bind(self, bind: Connection | Engine]):`, use `from __future__ import annotations` import to provide Python 3.8 support.